include/llbuild/Basic/FileInfo.h

```diff
@@ -88,10 +113,8 @@ struct FileInfo {
   bool isDirectory() const;
   
   bool operator==(const FileInfo& rhs) const {
-    return (device == rhs.device &&
-            inode == rhs.inode &&
-            size == rhs.size &&
-            modTime == rhs.modTime);
+    return (size == rhs.size &&
```

Comment:
Although this does make the behavior automatic, I'm not sure that changing operator== in this way is correct.   I think a user would rightly expect that comparing a FileInfo object would yield a deep comparison of all fields.   The contents comparison, as written here, should probably be a separate, explicit method.  That would also enable us to more cleanly implement @ddunbar 's comment about making this an opt-in feature.